modules/pbccs/main.nf

```diff
@@ -0,0 +1,49 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process PBCCS {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::pbccs=6.0.0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/pbccs:6.0.0--h9ee0642_2"
+    } else {
+        container "quay.io/biocontainers/pbccs:6.0.0--h9ee0642_2"
+    }
+
+    input:
+    tuple val(meta), path(bam), path(pbi), val(chunk_num), val(chunk_on)
+
+    output:
+    tuple val(meta), path("*.ccs.bam"), path("*.ccs.bam.pbi"), emit: bam
+    tuple path("*.ccs_report.txt"), path("*.ccs_report.json"), path ("*.zmw_metrics.json.gz"), emit: reports
```

Comment:
These should all be in separate channels (each file can be output with the `meta` tuple, so a pipeline dev can associate them together themselves if required)

Reply:
Understood, I update that.